imports/server/LoadBalancer.ts

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable no-underscore-dangle */
+import { IncomingMessage, ServerResponse } from 'http';
+import stream from 'stream';
+import { WebApp } from 'meteor/webapp';
+import HttpProxy from 'http-proxy';
+import WorkerPool, { Worker } from './WorkerPool';
+
+// This implements a simple multi-process, single-machine load-balancer.
+//
+// It serves as a drop-in replacement for the relevant subset of the
+// `meteorhacks:cluster` package we used (which was mostly just "load balance
+// across a pool of processes, so we can use all the cores on a machine").
+//
+// The coordinator runs in the main process, and may spawn several worker
+// processes (which will have behavior modulated by having CLUSTER_WORKER_ID set
+// in their environment).
+//
+// The coordinator intercepts HTTP and Websocket requests, and may dispatch them
+// to worker processes over `child_process` messaging.
+//
+// If CLUSTER_WORKERS_COUNT is set in the environment, it spawns that many child
+// processes, or if set to "auto", spawns as many children as the machine has
+// CPUs.
+//
+// Otherwise, it spawns no children, and the listeners simply propagate the
+// requests as middleware usually do, which causes the requests to be serviced
+// from the main process.
+export default class LoadBalancer {
+  proxy: HttpProxy;
+
+  workers: WorkerPool;
+
+  workerMapping: Record<string, { worker: Worker; lastUpdate: number; }>;
```

Comment:
There's no garbage collection of this, right?

Reply:
None at the moment (which matches the old upstream's behavior).  On errors, we [delete the entry](https://github.com/deathandmayhem/jolly-roger/pull/1166/files/dc423e5bc55568fa7959deeea46d38215ba3a2a4#diff-a74780b1bdb4e723525910156555ece545b6f8b0e739193495e470a6e8fa7ef4R78) so we'll pick a new backend at random, but we don't do any behavior like "if we haven't seen a request from this sockjs session in N minutes, drop the backend mapping".